File: snippets/js/react-native/src/read.ts
// ANCHOR: imports
import {
    Client,
    Filter,
    Kind,
    KindStandard,
    RelayUrl,
    ReqTarget,
} from "@nostrdevkit/nostr-sdk-react-native";
// ANCHOR_END: imports

// ANCHOR: stream
export async function streamEvents() {
    const client = new Client();
    await client.addRelay(RelayUrl.parse("wss://relay.damus.io"));
    await client.connect();

    const filter = new Filter()
        .kind(Kind.fromStd(KindStandard.TextNote))
        .limit(20n);
    const stream = await client.streamEvents(
        ReqTarget.auto([filter]),
        undefined,
        10_000,
    );

    while (true) {
        const item = await stream.next();
        if (!item) break;

        if (item.event) {
            console.log(`${item.relayUrl}: ${item.event.asJson()}`);
        } else if (item.error) {
            console.error(`${item.relayUrl}: ${item.error}`);
        }
    }

}
// ANCHOR_END: stream

// ANCHOR: fetch
export async function fetchEvents() {
    const client = new Client();
    await client.addRelay(RelayUrl.parse("wss://relay.damus.io"));
    await client.connect();

    const filter = new Filter()
        .kind(Kind.fromStd(KindStandard.TextNote))
        .limit(20n);
    const events = await client.fetchEvents(ReqTarget.auto([filter]), 10_000);

    for (const event of events) {
        console.log(event.asJson());
    }

}
// ANCHOR_END: fetch

File: snippets/swift/Sources/Read.swift
// ANCHOR: imports
import Foundation
import NostrSDK
// ANCHOR_END: imports

// ANCHOR: stream
func streamEvents() async throws {
    let client = Client()
    let relay = try RelayUrl.parse(url: "wss://relay.damus.io")
    _ = try await client.addRelay(url: relay)
    await client.connect()

    let filter = Filter()
        .kind(kind: Kind.fromStd(e: .textNote))
        .limit(limit: 20)
    let stream = try await client.streamEvents(
        target: ReqTarget.auto(filters: [filter]),
        timeout: 10.0
    )

    while let item = await stream.next() {
        if let event = item.event {
            print("\(item.relayUrl): \(try event.asJson())")
        } else if let error = item.error {
            print("\(item.relayUrl): \(error)")
        }
    }

}
// ANCHOR_END: stream

// ANCHOR: fetch
func fetchEvents() async throws {
    let client = Client()
    let relay = try RelayUrl.parse(url: "wss://relay.damus.io")
    _ = try await client.addRelay(url: relay)
    await client.connect()

    let filter = Filter()
        .kind(kind: Kind.fromStd(e: .textNote))
        .limit(limit: 20)
    let events = try await client.fetchEvents(
        target: ReqTarget.auto(filters: [filter]),
        timeout: 10.0
    )

    for event in events {
        print(try event.asJson())
    }

}
// ANCHOR_END: fetch
